docs/DevGuide/BuildOptimization.md

```diff
@@ -208,6 +208,34 @@ It is recommended to install and use
 to use the full KDE window system) to visualize the output -- the larger graphs
 produced by templight are inefficient to render in the graphviz format.
 
+## Clang profiling
+
+With `clang >= 9`, you can generate a flame graph of what the compiler is
+spending its time on, including specific function instantiations, class
+instantiations, source file parsing, debug info generation, and function
+optimization. To do so, compile your `cpp` of interest with `clang >= 9` and use
+the additional flag `-ftime-trace`. You can find the full compilation command
+for your `cpp` of interest by searching for it in the `compile_commands.json`
+file in your `build` directory. This will produce a
+`<your_cpp_target_name>.cpp.json` that you can then load into
+`chrome://tracing`. Because our code base utilizes many templates and some
```

Comment:
Maybe add a note that the `json` file (and the `.cpp.o` file also) may be very deep within the build directory. When I did this for the BBH executable, I found them in the
```
build/src/Evolution/Executables/GeneralizedHarmonic/CMakeFiles/EvolveGhBinaryBlackHole.dir/__/__/__/__/Evolution/Executables/GeneralizedHarmonic
```
directory.

Reply:
I'm happy to also be explicit and include a sentence like "For example, `EvolveGhBinaryBlackHole.cpp.o` and its `.json` are at <the directory you put above>." However, I can't figure out how to monospace multiple lines. I'd like to do something like this to respect the 80 character limit:
```
For example, `EvolveGhBinaryBlackHole.cpp.o` and its `.json` will be at
`build/src/Evolution/Executables/GeneralizedHarmonic/CMakeFiles/
EvolveGhBinaryBlackHole.dir/__/__/__/__/Evolution/Executables/
GeneralizedHarmonic`.
```
but the monospace formatting doesn't work. If you happen to know how to do this in markdown, I can add the whole file path.

Reply:
I'm not sure. I think it's ok to leave out.